doc/sphinx/language/core/basic.rst

```diff
@@ -146,6 +148,29 @@ Numbers
   The :cmd:`Number Notation` mechanism offers the user
   a way to define custom parsers and printers for :n:`@number`.
 
+  By default, numbers are interpreted as :n:`nat`\s, which is a unary
+  representation.  For example, :n:`3` is represented as `S (S (S O))`.  While
+  this is a convenient representation for doing proofs, computing with large
+  :n:`nat`\s can lead to stack overflows or running out of memory.  You can
+  explicitly specify a different interpretation to avoid this problem.  For
+  example, :n:`1000000%N` efficiently represents that natural number.  See
```

Comment:
```suggestion
  example, :n:`1000000%N` is a more efficient binary representation of that natural number.  See
```
But maybe rather take the more common Z as an example and link to https://coq.gitlab.io/-/coq/-/jobs/3977290672/artifacts/_build/default/doc/refman-html/language/coq-library.html#notations-for-integer-arithmetic

Reply:
The hyperlink points to info about notations for operations rather than for numbers, so I think not helpful here.  Too much detail.